lib/preprocessor.cpp

```diff
@@ -49,6 +49,12 @@ Directive::Directive(std::string _file, const int _linenr, const std::string &_s
     str(trim(_str))
 {}
 
+Directive::DirectiveToken::DirectiveToken(std::string _str, int _line, int _column) :
```

Comment:
Refactoring;

Preprocessor.h includes simplecpp.h so imho it would make sense to just provide 1 argument to the DirectiveToken constructor. A `const simplecpp::Token& tok`.

Similarly, the Directive constructor could take a `const simplecpp::Location& loc` instead of _file and _linenr. Unless it has to be constructed in other ways somewhere.